app/src/main/kotlin/com/github/oryanmat/trellowidget/activity/GeneralPreferenceFragment.kt

```diff
@@ -32,52 +32,83 @@ class GeneralPreferenceFragment : PreferenceFragment() {
-                colorPreference(R.string.pref_title_back_color_key).isEnabled = isChecked
+        when (key) {
+            getString(R.string.pref_update_interval_key) -> {
+                val preference = findPreference<ListPreference>(key)!!
+                val index = preference.findIndexOfValue(preference.value)
+                preference.summary = String.format(
+                    getString(
+                        R.string.pref_update_interval_value_desc
+                    ), preference.entries[index]
+                )
+            }
+
+            getString(R.string.pref_text_size_key) -> {
+                val preference = findPreference<ListPreference>(key)!!
+                val index = preference.findIndexOfValue(preference.value)
+                preference.summary = preference.entries[index]
+            }
+
+            getString(R.string.pref_back_color_key) -> {
+                val preference = findPreference<ColorPreference>(key)!!
+                preference.summary = String.format(COLOR_FORMAT, preference.color)
+            }
+
+            getString(R.string.pref_fore_color_key) -> {
+                val preference = findPreference<ColorPreference>(key)!!
+                preference.summary = String.format(COLOR_FORMAT, preference.color)
+            }
+
+            getString(R.string.pref_title_back_color_key) -> {
+                val preference = findPreference<ColorPreference>(key)!!
+                preference.summary = String.format(COLOR_FORMAT, preference.color)
+            }
+
+            getString(R.string.pref_title_fore_color_key) -> {
+                val preference = findPreference<ColorPreference>(key)!!
+                preference.summary = String.format(COLOR_FORMAT, preference.color)
+            }
+
+            getString(R.string.pref_title_use_unique_color_key) -> {
+                val preference = findPreference<SwitchPreference>(key)!!
+                with(preference) {
+                    summary = getString(R.string.pref_title_use_unique_color_desc)
+                    colorPreference(R.string.pref_title_fore_color_key).isEnabled = isChecked
+                    colorPreference(R.string.pref_title_back_color_key).isEnabled = isChecked
+                }
+            }
+
+            getString(R.string.pref_display_board_name_key) -> {
+                val preference = findPreference<SwitchPreference>(key)!!
+                preference.summary = getString(R.string.pref_display_board_name_desc)
             }
-        } else if (key == getString(R.string.pref_display_board_name_key)) {
-            val preference = findPreference(key) as SwitchPreference
-            preference.summary = activity.getString(R.string.pref_display_board_name_desc)
         }
     }
 
-    private fun colorPreference(@StringRes key: Int) = findPreference(getString(key)) as ColorPreference
+    private fun colorPreference(@StringRes key: Int) =
+        findPreference<ColorPreference>(getString(key))!!
```

Comment:
Just as an aside, I'd avoid using `!!` too often. Either use `?:` to return another value or throw an more descriptive error, or at least comment why it's safe to assume no null will pop up here.

Reply:
I would agree with you. I just added a try-catch to log the case.